tests/integration/devfile/cmd_devfile_status_test.go

```diff
@@ -38,325 +38,301 @@ var _ = Describe("odo devfile status command tests", func() {
 		helper.CommonAfterEach(commonVar)
 	})
 
-	Context("Verify devfile status works", func() {
-
-		It("Verify that odo component status correctly reports supervisord status", func() {
-
-			helper.Cmd("odo", "create", "--project", namespace, cmpName, "--devfile", helper.GetExamplePath("source", "devfiles", "nodejs", "devfile.yaml")).ShouldPass()
-
-			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
-
-			helper.Cmd("odo", "push", "-o", "json", "--project", namespace).ShouldPass()
-
-			session := helper.CmdRunner("odo", "component", "status", "-o", "json", "--project", namespace, "--follow")
-
-			helper.WaitForOutputToContain("supervisordStatus", 180, 10, session)
-
-			stdoutContents := string(session.Out.Contents())
-
-			entries, err := utils.ParseMachineEventJSONLines(stdoutContents)
-			Expect(err).NotTo(HaveOccurred())
-
-			// Verify supervisord status is present and correct
-			supervisordEntry := utils.GetMostRecentEventOfType(machineoutput.TypeSupervisordStatus, entries, true).(*machineoutput.SupervisordStatus)
-			count := 0
-			for _, status := range supervisordEntry.ProgramStatus {
-				if status.Program == string(common.DefaultDevfileRunCommand) {
-					Expect(status.Status).To(Equal("RUNNING"))
-				} else if status.Program == string(common.DefaultDevfileDebugCommand) {
-					Expect(status.Status).To(Equal("STOPPED"))
-				} else {
-					Fail(fmt.Sprintf("Unexpected program: %v", status.Program))
-				}
-				count++
+	//Function used to test context: "Verify URL status is correctly reported"
+	testCombo := func(ingress bool, route bool, name string) {
```

Comment:
`route` is not used anywhere.And if we remove this, it should reduce 2 extra calls to this function.